src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/DocumentsController.kt

```diff
@@ -31,24 +35,51 @@ class DocumentsController(
       is AuthorisableActionResult.Success -> Unit
     }
 
-    val documentsMetaDataResult = offenderService.getDocuments(crn)
-    val documentsMetaData = when (documentsMetaDataResult) {
+    val documentsFromApDeliusFlag = featureFlagService.getBooleanFlag("get-documents-from-ap-delius")
+    val documentsMetaData = when (val documentsMetaDataResult = getDocuments(retrieveDocumentsFromApDelius = documentsFromApDeliusFlag, crn)) {
       is AuthorisableActionResult.NotFound -> throw NotFoundProblem(crn, "Documents")
       is AuthorisableActionResult.Unauthorised -> throw ForbiddenProblem()
       is AuthorisableActionResult.Success -> documentsMetaDataResult.entity
     }
 
-    val documentMetaData = documentsMetaData.findDocument(documentId.toString())
-      ?: throw NotFoundProblem(documentId, "Document")
-
+    val documentFilename = getDocumentFileName(documentsMetaData, documentId)
     return ResponseEntity(
       StreamingResponseBody { outputStream ->
-        offenderService.getDocument(crn, documentId.toString(), outputStream)
+        offenderService.getDocument(
+          crn,
+          documentId.toString(),
+          outputStream,
+          isRetrieveDocumentFromApDelius = documentsFromApDeliusFlag,
+        )
       },
       HttpHeaders().apply {
-        put("Content-Disposition", listOf("attachment; filename=\"${documentMetaData.documentName}\""))
+        put("Content-Disposition", listOf("attachment; filename=\"$documentFilename\""))
       },
       HttpStatus.OK,
     )
   }
+
+  private fun getDocuments(retrieveDocumentsFromApDelius: Boolean, crn: String) = if (retrieveDocumentsFromApDelius) {
+    offenderService.getDocumentsFromApDeliusApi(crn)
+  } else {
+    offenderService.getDocumentsFromCommunityApi(crn)
+  }
+
+  private fun getDocumentFileName(documentsMetaData: Any, documentId: UUID): String? = when (documentsMetaData) {
+    is GroupedDocuments -> getDocumentFileName(documentsMetaData, documentId)
+    is List<*> -> getDocumentFileName(documentsMetaData, documentId)
```

Comment:
I guess Kotlin doesn't like using generic types here due to type erasure (E.g. `List<Document>`)?

If it was a long lived change we could maybe introduce an adapter that wraps the community api or ap-and-delius responses to provide the filename given an ID. Would then avoid the need to use the `Any` type and wildcard generics.

Reply:
makes sense, as we are going to strip out very soon will leave as-is for now. Sounds like that is okay